frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx

```diff
@@ -0,0 +1,185 @@
+import { actions as achievementsActions } from '~/bundles/achievements/store/achievements.js';
+import {
+    AchievementCard,
+    ActivityWidget,
+    ActivityWidgetColor,
+    Icon,
+    Loader,
+} from '~/bundles/common/components/components.js';
+import { IconColor } from '~/bundles/common/components/icon/enums/icon-colors.enum.js';
+import { IconName } from '~/bundles/common/components/icon/enums/icon-name.enum.js';
+import { DataStatus } from '~/bundles/common/enums/enums.js';
+import {
+    calculateTotal,
+    convertSecondsToHMS,
+    getLastWorkout,
+    metersToKilometers,
+} from '~/bundles/common/helpers/helpers.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+    useEffect,
+    useParams,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+import { actions as goalsActions } from '~/bundles/goals/store/goals.js';
+import {
+    PersonalDetails,
+    ProfileWorkoutItem,
+} from '~/bundles/profile/components/components.js';
+import { actions as userActions } from '~/bundles/users/store/users.js';
+import { actions as workoutsActions } from '~/bundles/workouts/store/workouts.js';
+
+const ZERO_VALUE = 0;
+const PublicProfile: React.FC = () => {
+    const { id } = useParams();
+
+    const dispatch = useAppDispatch();
+    const [isFollowed, setIsFollowed] = useState(false);
+    const NumericId = Number(id);
+
+    useEffect(() => {
+        void dispatch(userActions.getById(NumericId));
+        void dispatch(achievementsActions.getAchievementsByUserId(NumericId));
+        void dispatch(workoutsActions.getLastWorkoutsByUserId(NumericId));
+        void dispatch(goalsActions.getGoalsByUserId(NumericId));
+    }, [dispatch, NumericId]);
+
+    const { dataStatus: dataStatusUser, user } = useAppSelector(
+        ({ users }) => users,
+    );
+    const { dataStatus: dataStatusAchievements, achievements } = useAppSelector(
+        ({ achievements }) => achievements,
+    );
+    const { dataStatus: dataStatusWorkouts, workouts } = useAppSelector(
+        ({ workouts }) => workouts,
+    );
+    const { dataStatus: dataStatusGoals, goals } = useAppSelector(
+        ({ goals }) => goals,
+    );
```

Comment:
Maybe it's worth to create separate slice for this page?

Reply:
As I know for other pages we didn't create a separate slices, wouldn't it be confusing if I create it here?